labelbox/data/generator.py

```diff
@@ -43,15 +43,20 @@ def __init__(self, data: Iterable[Any], prefetch_limit=20, num_executors=4):
         self._data = ThreadSafeGen(self._data)
         self.completed_threads = 0
         # Can only iterate over once it the queue.get hangs forever.
+        self.multithread = False if num_executors == 1 else True
         self.done = False
-        self.num_executors = num_executors
-        self.threads = [
-            threading.Thread(target=self.fill_queue)
-            for _ in range(num_executors)
-        ]
-        for thread in self.threads:
-            thread.daemon = True
-            thread.start()
+
+        if self.multithread:
```

Comment:
move `self._data = ThreadSafeGen(self._data)` here. We don't need the lock if we aren't using threads. It is one less thing to go wrong.